sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala

```diff
@@ -1006,4 +1006,58 @@ class AnalysisSuite extends AnalysisTest with Matchers {
       checkAnalysis(plan, expect)
     }
   }
+
+  test("SPARK-22748: grouping_id() can only be used with GroupingSets/Cube/Rollup") {
+
+    val rollUpPlan = parsePlan("""select grouping__id from (
+                                 |select grouping__id from (
+                                 |select a, b, count(1), grouping__id from TaBlE2
+                                 | group by a, b with rollup))
+                                 |""".stripMargin)
```

Comment:
nit: could you re-format this test like this?
```
    assertAnalysisSuccess(parsePlan(
      """
        |SELECT grouping__id FROM (
        |  SELECT grouping__id FROM (
        |    SELECT a, b, count(1), grouping__id FROM TaBlE2
        |      GROUP BY a, b WITH ROLLUP
        |  )
        |)
      """.stripMargin))
```
 - I think we don't need intermediate values, e.g., `rollUpPlan`
 - Please use uppercases for SQL keywords

Reply:
ok